bench/BenchAll.hs

```diff
@@ -79,14 +87,57 @@ smallIntegerData = map fromIntegral intData
 largeIntegerData :: [Integer]
 largeIntegerData = map (* (10 ^ (100 :: Integer))) smallIntegerData
 
+{-# NOINLINE floatPosData #-}
+floatPosData :: [Float]
+floatPosData = map evenlyDistribute intData
+  where
+  evenlyDistribute :: Int -> Float
+  evenlyDistribute x = castWord32ToFloat $ increment * fromIntegral x
+  increment = castFloatToWord32 maxFinite `div` fromIntegral nRepl
 
-{-# NOINLINE floatData #-}
-floatData :: [Float]
-floatData = map (\x -> (3.14159 * fromIntegral x) ^ (3 :: Int)) intData
+{-# NOINLINE floatNegData #-}
+floatNegData :: [Float]
+floatNegData = map negate floatPosData
 
-{-# NOINLINE doubleData #-}
-doubleData :: [Double]
-doubleData = map (\x -> (3.14159 * fromIntegral x) ^ (3 :: Int)) intData
+{-# NOINLINE floatSpecials #-}
+floatSpecials :: [Float]
+floatSpecials = foldMap (const specials) [1..nRepl `div` length specials]
+  where
+  specials = [nan, infinity, negate infinity, 0 -0]
+
+{-# NOINLINE doublePosData #-}
+doublePosData :: [Double]
+doublePosData = map evenlyDistribute intData
+  where
+  evenlyDistribute :: Int -> Double
+  evenlyDistribute x = castWord64ToDouble $ increment * fromIntegral x
+  increment = (maximum - minimum) `div` fromIntegral nRepl
```

Comment:
I don't understand the idea here. Increment without offset? Does this result in only small `Double`s?

Reply:
When the mantissa is zero it signifies that the value is sub-normal.

Reply:
This is still suspicious.